### DISCUSSION TITLE
Finding types and available attributes

### DISCUSSION THREAD

**Discussion by aschoijett** [ArcadeData/arcadedb#515] [Q&A] [answered]
Hi,
Is there any way to return all the **types**, not the vertices, defined in the database that meet a certain condition (e.g., type starts with "A"?  

Likewise, is there any way to get all the attributes in a type that meet a certain condition (e.g., attribute is an integer)?

Many thanks!

Alex

**Comment by lvca** [comment, marked as answer]
Sure you can, you can query the available types with:

`select from schema:types`

This is not really documented, updating the docs asap...

**Reply by lvca** [reply]
Docs updated, will be online in 5 minutes.

**Reply by aschoijett** [reply]
Thank you so much.   On a related note, is there anything that I can read to learn how to define schemas for my types?  I think it's called schema-full but I cannot find how to do that from using a SQL statement to CREATE or ALTER a type and add my schema (much like you'd do using classical SQL DDL commands) -- and ideally do that from within HTTP REST API calls?

Many thanks!!

Alex

**Reply by lvca** [reply]
Take a look at the following SQL commands to manipulate the schema:
- create XX type
- alter type
- drop type
- create property
- alter property
- drop property

**Reply by aschoijett** [reply]
Many thanks!
